library/src/main/scala-new-collections/scala/scalajs/js/ArrayOps.scala

```diff
@@ -8,97 +8,2002 @@
+    (xs.jsSlice(0, idx), xs.jsSlice(idx))
+  }
+
+  /** Splits this array into two at a given position.
+   *
+   *  Note: `c.splitAt(n)` is equivalent to `(c.take(n), c.drop(n))`.
+   *
+   *  @param n
+   *    the position at which to split.
+   *  @return
+   *    a pair of arrays consisting of the first `n` elements of this array,
+   *    and the other elements.
+   */
+  def splitAt(n: Int): (js.Array[A], js.Array[A]) =
+    (take(n), drop(n))
+
+  /** A pair of, first, all elements that satisfy predicate `p` and, second,
+   *  all elements that do not.
+   */
+  def partition(p: A => Boolean): (js.Array[A], js.Array[A]) = {
+    val res1 = new js.Array[A]
+    val res2 = new js.Array[A]
+    var i = 0
+    val len = xs.length
+    while (i < len) {
+      val x = xs(i)
+      (if (p(x)) res1 else res2).push(x)
+      i += 1
+    }
+    (res1, res2)
+  }
+
+  /** Returns a new array with the elements in reversed order. */
+  def reverse: js.Array[A] = {
+    val len = xs.length
+    val res = new js.Array[A](len)
+    var i = 0
+    while (i < len) {
+      res(len - i - 1) = xs(i)
+      i += 1
+    }
+    res
+  }
+
+  /** An iterator yielding elements in reversed order.
+   *
+   *  Note: `xs.reverseIterator` is the same as `xs.reverse.iterator` but
+   *  implemented more efficiently.
+   *
+   *  @return
+   *    an iterator yielding the elements of this array in reversed order
+   */
+  def reverseIterator: scala.collection.Iterator[A] =
+    new js.ArrayOps.ReverseIterator[A](xs)
+
+  /** Selects all elements of this array which satisfy a predicate.
+   *
+   *  @param p
+   *    the predicate used to test elements.
+   *  @return
+   *    a new array consisting of all elements of this array that satisfy the
+   *    given predicate `p`.
+   */
+  def filter(p: A => Boolean): js.Array[A] = {
+    val res = new js.Array[A]
+    var i = 0
+    val len = xs.length
+    while (i < len) {
+      val x = xs(i)
+      if (p(x))
+        res.push(x)
+      i += 1
+    }
+    res
+  }
+
+  /** Selects all elements of this array which do not satisfy a predicate.
+   *
+   *  @param pred
+   *    the predicate used to test elements.
+   *  @return
+   *    a new array consisting of all elements of this array that do not
+   *    satisfy the given predicate `pred`.
+   */
+  def filterNot(p: A => Boolean): js.Array[A] =
+    filter(x => !p(x))
+
+  /** Sorts this array according to an Ordering.
+   *
+   *  The sort is stable. That is, elements that are equal (as determined by
+   *  `lt`) appear in the same order in the sorted sequence as in the original.
+   *
+   *  @see [[scala.math.Ordering]]
+   *
+   *  @param ord
+   *    the ordering to be used to compare elements.
+   *  @return
+   *    an array consisting of the elements of this array sorted according to
+   *    the ordering `ord`.
+   */
+  def sorted[B >: A](implicit ord: Ordering[B]): js.Array[A] =
+    new js.WrappedArray(xs).sorted(ord.asInstanceOf[Ordering[A]]).array
```

Comment:
Actually `WrappedArray` is already `@inline`, so we would just need to delegate to `WrappedArray`.

However, even with our kick-ass optimizer, the implementations inherited from the collections into `WrappedArray` are not as good as those we can directly write on `js.Array`, which means that `js.ArrayOps` would not be on par with `scala.collection.ArrayOps` if we do that. I did it for `sorted` because the original delegates to `java.util.Arrays.sort` anyway, which in Scala.js goes back to the collections (and also, it's really non-trivial code!).